### PR TITLE
DATAP-1471 - convert aggregationItem class to functional component

### PR DESCRIPTION
**Pull request by cdmh219**
Conversion of the AggregationItem component from a class to a function.
Updating of the unit tests to use React Testing Library instead of Jest.
Updating of Redux functionality to use useSelector hook instead of connect()

## Additions

- AggregationItem functional component and unit tests

## Removals

- AggregationItem class component and unit tests

## Changes

- The removeFilter() and addFilter() functions are now internal and local to the component, as mapDispatchToProps is no longer being used.

## Screenshots
Example
Screenshot 2024-07-29 at 9 18 09 AM
Screenshot 2024-07-29 at 9 17 50 AM

## Notes

-

## Todos

-

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: src/components/Filters/AggregationItem/AggregationItem.spec.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  fieldName: 'foo',
+  item: {
+    key: 'hole',
+    value: 'som',
+    doc_count: 100,
+    isDisabled: false,
+  },
+};
+
+const mockStore = configureMockStore();
```
Comment (flacoman91) [inline src/components/Filters/AggregationItem/AggregationItem.spec.js:16]: Sorry, i forgot to mention that we're trying to move off of mock store and setting up actual redux stores for testing.  

We have utilities to set up redux stores for testing in https://github.com/cfpb/ccdb5-ui/blob/main/src/testUtils/test-utils.js

You can set up a renderComponent function to set up tests.
Take a look at [HasNarrative.spec.js](https://github.com/cfpb/ccdb5-ui/blob/main/src/components/Filters/HasNarrative.spec.js) or [FilterPanelToggle.spec.js](https://github.com/cfpb/ccdb5-ui/blob/main/src/components/Filters/FilterPanelToggle.spec.js) as an example on how we set up tests.

Comment (cdmh219) [reply src/components/Filters/AggregationItem/AggregationItem.spec.js]: This has been updated accordingly

File: src/components/Filters/AggregationItem/AggregationItem.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+};
+
+const AggregationItem = ({ fieldName, item }) => {
+  const aggsState = useSelector(selectAggsState);
+  const queryState = useSelector(selectQueryState);
+  const dispatch = useDispatch();
+
+  const aggs = coalesce(aggsState, fieldName, []);
+  const filters = coalesce(queryState, fieldName, []);
+  const active =
```
Comment (flacoman91) [inline src/components/Filters/AggregationItem/AggregationItem.js:52]: booleans should have is `is` or `has` in front of them to make it clear when they're used.

https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/boolean-prop-naming.md

File: src/components/Filters/AggregationBranch.js [hunk 3]
```diff
@@ -7,7 +7,8 @@ import {
   slugify,
 } from '../../utils';
 import { removeMultipleFilters, replaceFilters } from '../../actions/filter';
-import AggregationItem from './AggregationItem';
+//import AggregationItem from './AggregationItem';
```
Comment (flacoman91) [inline src/components/Filters/AggregationBranch.js:10]: this can be deleted

File: src/components/Filters/AggregationItem/AggregationItem.spec.js [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+          requery: 'REQUERY_ALWAYS',
+          type: 'FILTER_CHANGED',
+        });
+
+        expect(replaceFiltersFn).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('removeFilter', () => {
+      /*test('handles product/issue filters', async () => {
```
Comment (flacoman91) [inline src/components/Filters/AggregationItem/AggregationItem.spec.js:259]: does this test need restoring?

**Review by flacoman91 — CHANGES_REQUESTED** [review]

File: src/components/Filters/AggregationItem/AggregationItem.js (hunk 2 above)
Comment (cdmh219) [reply src/components/Filters/AggregationItem/AggregationItem.js]: I usually do it for functions and methods, but have updated here accordingly.

File: src/components/Filters/AggregationBranch.js (hunk 3 above)
Comment (cdmh219) [reply src/components/Filters/AggregationBranch.js]: This has been taken out

**Review by flacoman91 — APPROVED** [review]

**Pull request merged**